### PR TITLE
Using jxl::extras::Encoder in djxl_ng

### PR DESCRIPTION
**Pull request by mo271**
This replaces the direct call to jxl::extras::EncodeImagePNM.

### CODE REVIEW

File: tools/djxl_ng_main.cc [hunk 1]
```diff
@@ -425,52 +425,44 @@ int main(int argc, char** argv) {
...
-        JXL_WARNING("For color images, the filename should end with .ppm.\n");
-      }
-      if (num_channels == 3 && ppf.info.num_color_channels == 1) {
-        JXL_WARNING(
-            "For grayscale images, the filename should end with .pgm.\n");
-      }
-      if (ppf.info.bits_per_sample > 16) {
-        JXL_WARNING("PPM only supports up to 16 bits per sample");
-      }
+    if (!jxl::extras::SelectFormat(encoder->AcceptedFormats(), ppf.info,
```
Comment (sboukortt) [inline tools/djxl_ng_main.cc:439]: What is `format` used for outside of `DecompressJxlToPackedPixelFile`? Do we need this?

File: tools/djxl_ng_main.cc [hunk 2]
```diff
@@ -425,52 +425,44 @@ int main(int argc, char** argv) {
...
+    }
+    jxl::extras::EncodedImage encoded_image;
+    JXL_RETURN_IF_ERROR(encoder->Encode(ppf, &encoded_image));
+    if (ppf.frames.size() != encoded_image.bitstreams.size()) {
+      fprintf(stderr,
+              "didn't receive expected number of frames: %" PRIuS
+              " (expected) versus %" PRIuS "(received)\n",
+              ppf.frames.size(), encoded_image.bitstreams.size());
+      return EXIT_FAILURE;
     }
```
Comment (sboukortt) [inline tools/djxl_ng_main.cc:452]: I am not completely sure it’s for this function to check that `Encode` returned the correct number of bitstreams (which could be 1 if the format supports animations). I think I would favour trusting `Encode` to return an appropriate number of bitstreams or an error if one occurs, and replacing the instances of `ppf.frames.size()` below with `encoded_image.bitstreams.size()`. What do you think?

File: tools/djxl_ng_main.cc [hunk 3]
```diff
@@ -425,52 +425,44 @@ int main(int argc, char** argv) {
...
-      if (ppf.info.bits_per_sample > 16) {
-        JXL_WARNING("PPM only supports up to 16 bits per sample");
-      }
+    if (!jxl::extras::SelectFormat(encoder->AcceptedFormats(), ppf.info,
+                                   &format)) {
+      fprintf(stderr, "SelectFormat failed\n");
+      return EXIT_FAILURE;
+    }
+    jxl::extras::EncodedImage encoded_image;
+    JXL_RETURN_IF_ERROR(encoder->Encode(ppf, &encoded_image));
```
Comment (sboukortt) [inline tools/djxl_ng_main.cc:445]: If this fails, `JXL_RETURN_IF_ERROR` will return `false`, which will mean a return code equal to `EXIT_SUCCESS`.

File: tools/djxl_ng_main.cc (hunk 1 above)
Comment (mo271) [reply tools/djxl_ng_main.cc]: We don't need it; removed.

File: tools/djxl_ng_main.cc (hunk 3 above)
Comment (mo271) [reply tools/djxl_ng_main.cc]: changed the check

**Review by sboukortt — APPROVED** [review]

**Pull request merged**